src/output.js

```diff
@@ -0,0 +1,142 @@
+const { groupBy, find, get, findLast } = require('lodash')
+const { IncomingWebhook } = require('@slack/webhook')
+const github = require('@actions/github')
+const { readFileSync, readdirSync } = require('fs')
+const { join } = require('path')
+
+/** @type {string} */
+const homeDir = get(process.env, 'HOME', '')
+/** @type {string} */
+const githubRepo = get(process.env, 'GITHUB_REPOSITORY', '')
+/** @type {string} */
+const githubSHA = get(process.env, 'GITHUB_SHA', '')
+
+/**
+ * @param {{type: 'slack', status: number, slackWebhookUrl: string, githubToken: string, staticDistDir: string}} params
+ */
+async function run({ type, ...args }) {
+  try {
+    if (type === 'slack') {
+      await slackNotification(args)
+    } else {
+      console.log('Unknown output type: ', type)
+    }
+  } catch (e) {
+    throw e
+  }
+}
+
+/**
+ * @param {{status: number, slackWebhookUrl: string, githubToken: string, staticDistDir: string}} params
+ * @return {Promise<*>}
+ */
+async function slackNotification({ status, slackWebhookUrl, githubToken, staticDistDir = '.lighthouseci' }) {
+  const conclusion = status === 0 ? 'success' : 'failure'
+  const webhook = new IncomingWebhook(slackWebhookUrl)
+  const color = status === 0 ? 'good' : 'danger'
+  const staticDistDirPath = join(homeDir, staticDistDir)
+  const LHRNameFromPath = getLHRNameFromPath(staticDistDirPath)
+
+  const assertionResults = readFileSync(join(staticDistDirPath, 'assertion-results.json'))
+  const groupedResults = groupBy(JSON.parse(assertionResults.toString()), 'url')
+
+  const attachments = Object.values(groupedResults).map(groupedResult => {
```

Comment:
This seems complex enough to split into its own function tbh. Like 
```javascript
function assertionResultsFormatterForSlack()
```
But with a better name...